### PR TITLE
Java conditionals and java semicolon linter

### PR DESCRIPTION
**Pull request by cjoshea9**
Adds test cases and implementation for java conditional statements.

Also added a java linter function to add semicolons at the end of lines.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -61,3 +64,22 @@ def run_bash_javascript(input_code):
         out, error = process.communicate()
         string_output = out.decode('utf-8')
         return json.loads(string_output)
+
+"""
+Add semicolons to the end of java lines
+"""
+def java_linter(output_code):
```
Comment (jackdavidweber) [inline main.py:71]: should this be in main.py? Feel like it should be in one of the java folders?

File: main.py [hunk 2]
```diff
@@ -61,3 +64,22 @@ def run_bash_javascript(input_code):
...
+
+    if '\n' not in output_code:
+        return output_code + ";"
+
+    #Add semicolons to every line
+    output_code = output_code.replace("\n", ";\n")
+    output_code += ";"
+    
+    #Remove them where they shouldn't be
```
Comment (jackdavidweber) [inline main.py:80]: Are we missing any cases?

File: test/test_code_to_gast_conditionals.py [hunk 3]
```diff
@@ -45,10 +49,12 @@ def test_else(self):
             }
         self.assertEqual(expected_gast, py_main.py_to_gast(py_code))
         self.assertEqual(expected_gast, js_main.js_to_gast(js_code))
+        self.assertEqual(expected_gast, java_main.java_to_gast(java_code))
    
     def test_elif(self):
         py_code = 'if (1):\n\tprint("1 is true")\nelif (2):\n\tprint("2 is true")\n\tprint("second line")\n'
         js_code = 'if (1) {\n\tconsole.log("1 is true")\n} else if (2) {\n\tconsole.log("2 is true")\n\tconsole.log("second line")\n}'
+        java_code = 'if (1) {System.out.println("1 is true");} else if (2) {System.out.println("2 is true"); System.out.println("second line");}'
```
Comment (swalsh15) [inline test/test_code_to_gast_conditionals.py:57]: Is indenting supported for java?

File: main.py [hunk 4]
```diff
@@ -61,3 +64,22 @@ def run_bash_javascript(input_code):
         out, error = process.communicate()
         string_output = out.decode('utf-8')
         return json.loads(string_output)
+
+"""
+Add semicolons to the end of java lines
+"""
+def java_linter(output_code):
+
+    if '\n' not in output_code:
```
Comment (swalsh15) [inline main.py:73]: The if negative pattern is a little confusing - maybe add a comment or reverse the pattern

File: test/test_conditionals.py [hunk 5]
```diff
@@ -6,34 +6,46 @@ def test_if(self):
         js_code = 'if (1) {\n\tconsole.log("This is true")\n}'
         py_code = 'if (1):\n\tprint("This is true")'
         bash_code = 'if [[ 1 ]]; then\n\techo "This is true"\nfi'
+        java_code = 'if (1) {\n\tSystem.out.println("This is true");\n}'
         self.assertEqual(py_code, main.main(js_code, 'js', 'py'))
         self.assertEqual(js_code, main.main(py_code, 'py', 'js')) 
         self.assertEqual(bash_code, main.main(py_code, 'py', 'bash'))
         self.assertEqual(bash_code, main.main(js_code, 'js', 'bash'))
+        self.assertEqual(java_code, main.main(js_code, 'js', 'java'))
```
Comment (jackdavidweber) [inline test/test_conditionals.py:14]: These tests are starting to get confusing. I created a [taiga ticket](https://tree.taiga.io/project/jackdavidweber-cjs-capstone/us/148?no-milestone=1) to break up all tests by language and I think this serves as a good example for the need.

Comment (jackdavidweber) [reply test/test_conditionals.py:14]: (To be dealt with in another PR)

File: main.py (hunk 1 above)
Comment (cjoshea9) [reply main.py:71]: Yeah, I agree but I don't know where exactly makes sense

File: main.py (hunk 2 above)
Comment (cjoshea9) [reply main.py:80]: Might be, but not that I can think of

File: main.py (hunk 4 above)
Comment (cjoshea9) [reply main.py:73]: Okay

File: test/test_code_to_gast_conditionals.py (hunk 3 above)
Comment (cjoshea9) [reply test/test_code_to_gast_conditionals.py:57]: Yes for gast_to_code, but it is not necessary when testing code to gast. Java parser works the same way with and without indentation

File: main.py (hunk 1 above)
Comment (augerm) [reply main.py:71]: Can this live in gast_to_code helpers? It seems like JavaScript should have this same problem.

File: java/gast_to_code/gast_to_code_java.py [hunk 6]
```diff
@@ -11,7 +12,21 @@ def handle_bool(gast):
             return "false"
 
     def handle_if(gast, lvl=0):
-        pass
+        test = router.gast_to_code(gast["test"], "js")
```
[the remark this answers is not in the document]
Comment (cjoshea9) [reply java/gast_to_code/gast_to_code_java.py:15]: Yes

File: java/code_to_gast/java_router.py [hunk 7]
```diff
@@ -2,13 +2,14 @@
...
 import java.code_to_gast.java_assign as java_assign
+import java.code_to_gast.java_conditional as java_conditional
 
 def node_to_gast(node):
     #base cases
     if type(node) == javalang.tree.Literal:
         # node.value is stored as string in java AST
         if node.value.isnumeric():
-            return {"type": "num", "value": node.value}
+            return {"type": "num", "value": int(node.value)}
```
Comment (augerm) [inline java/code_to_gast/java_router.py:12]: Is the `int()` cast going to be a problem with decimals?

Comment (cjoshea9) [reply java/code_to_gast/java_router.py:12]: We are currently only handling ints, and have a ticket to change that to handle both

Comment (augerm) [reply java/code_to_gast/java_router.py:12]: Gotcha. Can you add a comment like TODO(<username>) or better TODO(<taiga ticket #>)

**Pull request merged**